cpp/src/arrow/util/bit_util.h

```diff
@@ -116,7 +119,7 @@ static constexpr uint8_t kBytePopcount[] = {
 
 // Returns the ceil of value/divisor
 constexpr int64_t CeilDiv(int64_t value, int64_t divisor) {
-  return value / divisor + (value % divisor != 0);
+  return SETNE(value) + (value - SETNE(value)) / divisor;
```

Comment:
This seems needlessly obscure to me. Would the following work instead?
```cpp
  return (value == 0) ? 0 : 1 + (value - 1) / divisor;
```

Reply:
@pitrou Thanks a lot for your comments.

Your suggested code really works and is much clearer. 
I am just concerned that it will create a branch in the instruction and affect performance. 

What is your opinion about this?

Reply:
Do you have any benchmark where this is important?

Reply:
I have added a simple benchmark. The result shows that there are some overhead with the tertiary operator:

The setnz version:
CeilDev                                       554 ns          555 ns      1262719 items_per_second=100.969M/s

The tertiary operation version:
CeilDev                                       642 ns          643 ns      1000000 items_per_second=87.1129M/s

Reply:
Sorry, there was a misunderstanding. I don't think we care about micro-benchmarking `CeilDiv`. What I was asking was whether there was an existing benchmark or workload where you saw a difference.

Reply:
@pitrou Sorry I misunderstood your point. The microbenchmark has been removed. 

I ran bit_util_benchmark, and found no notable change:

Benchmark    setne impl    tertiary impl
VisitBits/8192  116.806M/s      116.342M/s
VisitBitsUnrolled/8192  319.859M/s      317.137M/s
BitmapWriter/8192       81.4389M/s      80.8341M/s
FirstTimeBitmapWriter/8192      87.3256M/s      87.0754M/s
GenerateBits/8192       81.1347M/s      81.0713M/s
GenerateBitsUnrolled/8192       138.403M/s      128.731M/s
CopyBitmapWithoutOffset/8192    38.871G/s       39.4095G/s
CopyBitmapWithOffset/8192       638.879M/s      638.865M/s
BenchmarkBitmapAnd/32768/0      4.78202G/s      4.77886G/s
BenchmarkBitmapAnd/131072/0     4.77979G/s      4.78636G/s
BenchmarkBitmapAnd/32768/0      4.77655G/s      4.77886G/s
BenchmarkBitmapAnd/131072/0     4.78616G/s      4.77806G/s
BenchmarkBitmapAnd/32768/1      17.5473M/s      17.5703M/s
BenchmarkBitmapAnd/131072/1     17.5653M/s      17.5669M/s
BenchmarkBitmapAnd/32768/2      17.699M/s       17.6438M/s
BenchmarkBitmapAnd/131072/2     17.7274M/s      17.6017M/s
BenchmarkBitmapVisitBitsetAnd/32768/0   23.1401M/s      23.1495M/s
BenchmarkBitmapVisitBitsetAnd/131072/0  23.1276M/s      23.1548M/s
BenchmarkBitmapVisitBitsetAnd/32768/0   23.1153M/s      23.1555M/s
BenchmarkBitmapVisitBitsetAnd/131072/0  23.1378M/s      23.1376M/s
BenchmarkBitmapVisitBitsetAnd/32768/1   23.0758M/s      23.1538M/s
BenchmarkBitmapVisitBitsetAnd/131072/1  23.1484M/s      23.1581M/s
BenchmarkBitmapVisitBitsetAnd/32768/2   23.1425M/s      23.1426M/s
BenchmarkBitmapVisitBitsetAnd/131072/2  23.1366M/s      23.1431M/s
BenchmarkBitmapVisitUInt8And/32768/0    677.893M/s      679.319M/s
BenchmarkBitmapVisitUInt8And/131072/0   681.277M/s      679.301M/s
BenchmarkBitmapVisitUInt8And/32768/0    684.623M/s      681.35M/s
BenchmarkBitmapVisitUInt8And/131072/0   678.489M/s      678.842M/s
BenchmarkBitmapVisitUInt8And/32768/1    301.361M/s      302.237M/s
BenchmarkBitmapVisitUInt8And/131072/1   303.976M/s      304.196M/s
BenchmarkBitmapVisitUInt8And/32768/2    300.889M/s      301.587M/s
BenchmarkBitmapVisitUInt8And/131072/2   304.244M/s      304.374M/s
BenchmarkBitmapVisitUInt64And/32768/0   2.71933G/s      2.71748G/s
BenchmarkBitmapVisitUInt64And/131072/0  2.70706G/s      2.72068G/s
BenchmarkBitmapVisitUInt64And/32768/0   2.70293G/s      2.69589G/s
BenchmarkBitmapVisitUInt64And/131072/0  2.70683G/s      2.72018G/s
BenchmarkBitmapVisitUInt64And/32768/1   1.58441G/s      1.58869G/s
BenchmarkBitmapVisitUInt64And/131072/1  1.74613G/s      1.75083G/s
BenchmarkBitmapVisitUInt64And/32768/2   1.59468G/s      1.59617G/s
BenchmarkBitmapVisitUInt64And/131072/2  1.74953G/s      1.74816G/s